src-self-hosted/main.zig

```diff
@@ -772,7 +772,7 @@ async fn fmtPath(fmt: *Fmt, file_path_ref: []const u8, check_mode: bool) FmtErro
     };
     defer fmt.loop.allocator.free(source_code);
 
-    var tree = std.zig.parse(fmt.loop.allocator, source_code) catch |err| {
+    var tree = std.zig.parseAndTurnABlindEyeToInvalidWhitespace(fmt.loop.allocator, source_code) catch |err| {
```

Comment:
If we ignore all invalid tokens, then we'll read UTF-16 encoded documents wrong in a very misleading way. (I can try to come up with examples, but in short it would be a Bad Idea™.)

It might make more sense to have a preprocessing step where we translate UTF-16 and any other encoding into UTF-8, and at the same time turn `"\t"` into `"    "` and normalize newlines. Then instead of trying to resolve the whitespace problems addressed in this PR "in-place" by ignoring them, we properly fix them by replacing them with other characters.

I think I'm describing a better long-term solution. I'm convinced that this PR is not the way forward.